lib/services/itmstransporter-service.ts

```diff
@@ -0,0 +1,88 @@
+///<reference path="../.d.ts"/>
+"use strict";
+import * as path from "path";
+import * as temp from "temp";
+let md5 = require("cryptojs").Crypto.MD5;
+
+export class ITMSTransporterService implements IITMSTransporterService {
+	private _itmsTransporterPath: string = null;
+
+	constructor(private $childProcess: IChildProcess,
+		private $devicePlatformsConstants: Mobile.IDevicePlatformsConstants,
+		private $errors: IErrors,
+		private $fs: IFileSystem,
+		private $platformService: IPlatformService,
+		private $staticConfig: IStaticConfig,
+		private $sysInfo: ISysInfo) { }
+
+	public upload(appId: string, username: string, password: string, mobileProvisionIdentifier?: string, codeSignIdentity?: string): IFuture<void> {
+		return (() => {
+			temp.track();
+			let itmsTransporterPath = this.getITMSTransporterPath().wait(),
+				ipaFileName = "app.ipa",
+				itmsDirectory = temp.mkdirSync("itms-"),
+				innerDirectory = path.join(itmsDirectory, "mybundle.itmsp"),
+				ipaFileLocation = path.join(innerDirectory, ipaFileName),
+				platform = this.$devicePlatformsConstants.iOS,
+				forDevice = true,
+				iosBuildConfig: IiOSBuildConfig = {
+					buildForDevice: forDevice,
+					mobileProvisionIdentifier: mobileProvisionIdentifier,
+					codeSignIdentity: codeSignIdentity
+				};
+
+			this.$fs.createDirectory(innerDirectory).wait();
+
+			this.$platformService.preparePlatform(platform).wait();
```

Comment:
:cry: 
Let's add `--` parameter for path to `.ipa`, so the user can skip the build